docs/function_guide.rst

```diff
@@ -0,0 +1,264 @@
+.. autofunction:: sendrecv_mgr_worker_msg
+
+This function performs both of the previous functions in a single statement. Its
+usage typically resembles::
+
+    tag, Work, calc_in = sendrecv_mgr_worker_msg(libE_info['comm'], local_H_out[selected_IDs])
+    if tag in [STOP_TAG, PERSIS_STOP]:
+        cleanup()
+        break
+
+Once the persistent generator's loop has been broken, it should return with an additional tag::
+
+    return local_H_out, persis_info, FINISHED_PERSISTENT_GEN_TAG
+
+See :doc:`calc_status<data_structures/calc_status>` for more information about
+the message tags.
+
+Examples of normal and persistent generator functions
+can be found :doc:`here<examples/gen_funcs>`.
+
+Simulator Functions
+===================
+
+As described in the :ref:`API<api_sim_f>`, the ``sim_f`` is called by a
+libEnsemble worker via a similar interface to the ``gen_f``::
+
+    out = sim_f(H[sim_specs['in']][sim_ids_from_allocf], persis_info, sim_specs, libE_info)
+
+In practice, most ``sim_f`` function definitions written by users resemble::
+
+    def my_simulator(H, persis_info, sim_specs, libE_info):
+
+Where :doc:`sim_specs<data_structures/sim_specs>` is a
+dictionary containing pre-defined parameters for the ``sim_f``, and the other
+parameters serve similar purposes to those in the ``gen_f``.
+
+The pattern of setting up a local ``H``, parsing out parameters from
+``sim_specs``, performing calculations, and returning the local ``H``
+with ``persis_info`` should be familiar::
+
+    batch_size = sim_specs['user']['batch_size']
+    local_H_out = np.zeros(batch_size, dtype=sim_specs['out'])
+
+    ... # Perform simulation calculations
+
+    return local_H_out, persis_info
+
+Descriptions of included simulator functions can be found :ref:`here<examples/sim_funcs>`.
+
+The :ref:`Simple Sine tutorial<tutorials/local_sine_tutorial>` is an
+excellent introduction for writing simple user functions and using them
+with libEnsemble.
+
+Executor
+--------
+
+libEnsemble's Executor is commonly used within simulator functions to launch
+and monitor applications. An excellent overview is already available
+:doc:`here<executor/overview>`.
+
+See the :doc:`Executor with Electrostatic Forces tutorial<tutorials/executor_forces_tutorial>`
+for an additional example to try out.
+
+Allocation Functions
+====================
+
+Although the included allocation functions, or ``alloc_f``'s are sufficient for
+most users, those who want to fine-tune how data is passed to their ``gen_f``
+and ``sim_f`` can write their own. The ``alloc_f`` is unique since it is called
+by the libEnsemble's manager instead of a worker.
+
+Most ``alloc_f`` function definitions written by users resemble::
+
+    def my_allocator(W, H, sim_specs, gen_specs, alloc_specs, persis_info):
+
+Where :doc:`W<data_structures/worker_array>` is an array containing information
+about each worker's state, and ``H`` is the *entire* History array available to
```

Comment:
The H sent to the allocation function is the trimmed H, it only contains rows intialized by the generator. So I guess its pretty much the entire useful array.